### PR TITLE
Add proposal draft

### PR DESCRIPTION
**Pull request by lwm**
Follow on from #1 (closed due to resetting this repo - experimenting).

### CODE REVIEW

**Comment by lwm** [comment]
Notes on the current status:

  - Missing something more concrete for the 'How will your project solve this problem?' question.
  - I want to highlight the opportunity to provide the 'narrative' documentation alongside the dynamically generated technical. I am becoming more interested in this idea (ala Stripe documentation). Not sure if I made that point well.
  - Reviewers have told me that including client libraries seems like the project wants to deliver too much. The project only asks for a prototype in the end. They could possibly be removed to focus more on the platform.
  - Th 'Important milestones' section should be more concrete. If I was to start tomorrow, I should exactly what I want to do. I need to work on that.

File: _drafts/okfn.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  web API creators can focus on delivering the narrative.
+
+There are many schema formats available to web API creators today. This tool
+will not be limited to a single schema format. The Core API library offers a
+translation between many different schemas and a single representation. This
+single representation will be used by the tool to provide the advantages of
+schema-driven documentation and client libraries for any web APIs that exposes
+a schema.
+
+# Who is the target audience? How would they benefit from this project?
```
Comment (tomchristie) [inline _drafts/okfn.md]: The first and last paragraphs of this section are the core points. You probably don't need the rest.

File: _drafts/okfn.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  - https://github.com/lwm?tab=overview&period=monthly
+
+I also maintain a OSS related blog:
+  - http://lwm.github.io/
+
+# Project title?
+
+coreapi-doc, a web API documentation tool.
+
+# What problem do you want to solve with your project?
```
Comment (tomchristie) [inline _drafts/okfn.md]: You probably could cut down this section a little, if wanted.

The core message is simply that writing and maintaining API documentation and client libraries is time consuming and expensive. Documentation can easily fall out of sync with the API, and maintaining client libraries across several languages requires a huge investment.

File: _drafts/okfn.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+I contribute to various other OSS projects that I use (scroll to 'Contribution activity'):
+  - https://github.com/lwm?tab=overview&period=monthly
+
+I also maintain a OSS related blog:
+  - http://lwm.github.io/
+
+# Project title?
+
+coreapi-doc, a web API documentation tool.
```
Comment (tomchristie) [inline _drafts/okfn.md]: Perhaps us a proper name here, eg. "Core API Doc."

File: _drafts/okfn.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  application interface to provide high quality integration with their
+  application code.
+
+# What was your motivation?
+
+New developments in the web API community are providing possibilities to help
+solve the difficulties described above. I'd like to be a part of providing
+those solutions.
+
+# How will your project solve this problem?
```
Comment (tomchristie) [inline _drafts/okfn.md]: Like this section, yup.

File: _drafts/okfn.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+This project can democratize that power.
+
+This project can significantly reduce the cost, effort and maintenance for
+creators who wish to deliver documentation and client libraries. This powerful
+tool will be open for the community to use and improve.
+
+# Have you ever been working on the idea? If so, briefly describe the current state and explain the new feature.
+
+I have been working on the design, concept and proposal of this project. Tom
+Christie, the creator of the Core API library, has kindly provided an
```
Comment (tomchristie) [inline _drafts/okfn.md]: Perhaps phrase this more along the lines that we've been collaborating on the idea or something. "Has provided an introduction to the topic"might be mis-interpreted - the driving force here is you. 😄

File: _drafts/okfn.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+introduction to the topic and deems it a worthwhile project to undertake.
+
+# How many hours will you probably need during a period of 6 months for the implementation of the project idea?
+
+I would estimate to work full time (~35 hours per week) on this project for the
+duration of the 6 months. I can roughly estimate the following times:
+
+- documentation tool: 2/3 months
```
Comment (tomchristie) [inline _drafts/okfn.md]: Minor point - I'd phrase 2/3 in plain language, eg. Between 2 and 3 months.
On first reading it looks like two thirds of a month to me.

File: _drafts/okfn.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+- 3/4 client libraries: 1 month per library
```
Comment (tomchristie) [inline _drafts/okfn.md]: Just "Client libraries: 1 month per library" might scan a little more cleanly?

**Comment by tomchristie** [comment]
Looking good!

File: _drafts/okfn.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+consuming and expensive. Documentation is expected to explain every single
+aspect of what the web API offers, both technical and narrative, and can easily
+fall out of sync with the web API. Maintaining client libraries across several
+languages requires a huge investment.
+
+# What was your motivation?
+
+New developments in the web API community are providing possibilities to help
+solve the difficulties described above. I'd like to be a part of providing
+those solutions.
```
Comment (grady-lad) [inline _drafts/okfn.md]: Maybe `unifying those solutions` instead of 'providing those solutions' ?

File: _drafts/okfn.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+huge audience. This project can significantly reduce the cost, effort and
+maintenance for creators who wish to deliver documentation and client
+libraries. This powerful tool will be open for the community to use and
+improve.
+
+# Have you ever been working on the idea? If so, briefly describe the current state and explain the new feature.
+
+I have been working on the design, concept and proposal of this project. I have
+been collaborating on the idea with Tom Christie, the creator of the Core API
+library.
```
Comment (grady-lad) [inline _drafts/okfn.md]: Would it to be possible to elaborate more on `collaborating the idea with Tom Christie` ?

File: _drafts/okfn.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+    - Add client documentation for the existing Python client to the rendered output.
+    - Add ability to combine narrative documentation with rendered output.
+  - Second milestone:
+    - @TODO
+  - Third milestone:
+    - Create and integrate new client libraries
+
+Note. The various client libraries will be developed after the core project is
+in place, reducing the risk of delivering the project.
+
```
Comment (grady-lad) [inline _drafts/okfn.md]: So you will focus on one client library at first ?

Comment (lwm) [inline _drafts/okfn.md]: The client libraries are the last thing to be done. There is an existing python client which I will use to integrate (mentioned in first milestone).

File: _drafts/okfn.md (hunk 9 above)
Comment (lwm) [inline _drafts/okfn.md]: I also had this commit: [1a83142](https://github.com/lwm/lwm.github.io/pull/3/commits/1a83142127bd4329a5f2892d14a784076e240663#diff-2c431a17ae1eae1fe04cba374973963fR106) but ended up reducing it. In the end, I think I will follow the point made in https://github.com/lwm/lwm.github.io/pull/3#discussion_r76761708.

File: _drafts/okfn.md (hunk 8 above)
Comment (lwm) [inline _drafts/okfn.md]: Sounds better!

**Comment by lwm** [comment]
Thanks for reviewing folks!!!

**Comment by domenkozar** [comment]
How is this different to http://raml.org/ or http://swagger.io/?

**Comment by tomchristie** [comment]
RAML and Swagger are API schemas, that a developer can use to describe the API.
This is a documentation tool (& client libraries) that is driven by an API schema.

It can be used to provide user-friendly documentation or interact to with any API that has a RAML or a Swagger schema.

**Pull request merged**